modules/seqexec/server/src/main/scala/seqexec/server/EpicsUtil.scala

```diff
@@ -245,4 +246,15 @@ object EpicsUtil {
   def smartSetParam[A: Eq](v: A, get: => Option[A], set: SeqAction[Unit]): SeqAction[Unit] =
     if(get =!= v.some) set else SeqAction.void
 
+  def countdown[F[_]: Effect](total: Time, remT: F[Option[Time]],
+                              obsState: F[Option[CarStateGeneric]]): Stream[F, Progress] =
+    ProgressUtil.fromFOption(_ => (remT, obsState).mapN { case (rem, st) =>
+      for{
+        c <- rem
+        s <- st
+        if s.isBusy
+      } yield Progress(if(total>c) total else c, RemainingTime(c))
+    }).dropWhile(_.remaining.self.value === 0.0) // drop leading zeros
+      .takeThrough(_.remaining.self.value > 0.0) // drop all tailing zeros but the first one
+
```

Comment:
It is only active while the exposure is in progress, removes leading and tailing zeros, and never produces a `Progress`  where the remaining time is higher than the total.

Reply:
Neat